src/LibraryManager.Vsix/Json/Completion/LibraryIdCompletionEntry.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.Web.Editor.Completion;
+using Microsoft.VisualStudio.Text;
+using System.Diagnostics;
+
+namespace Microsoft.Web.LibraryManager.Vsix
+{
+    internal class LibraryIdCompletionEntry : SimpleCompletionEntry
+    {
+        internal int Major { get; private set; }
+
+        internal int Minor { get; private set; }
+
+        internal int Patch { get; private set; }
+
+        internal bool IsValid { get; private set; }
+
+        public LibraryIdCompletionEntry(string displayText, string insertionText, ImageMoniker moniker, IIntellisenseSession session)
+            : base(displayText, insertionText, moniker, session)
+        {
+        }
+
+        public LibraryIdCompletionEntry(string displayText, string insertionText, string description, ImageMoniker moniker, ITrackingSpan span, IIntellisenseSession session, bool versionSpecific = false)
+            : base(displayText, insertionText, description, moniker, span, session)
+        {
+            if (versionSpecific)
+            {
+                SetVersion(insertionText);
+            }
+        }
+
+        protected override int InternalCompareTo(CompletionEntry other)
+        {
+            if (other == null)
+            {
+                return 1;
+            }
+
+            LibraryIdCompletionEntry otherEntry = other as LibraryIdCompletionEntry;
+
+            if (IsValid)
+            {
+                int result = -Major.CompareTo(otherEntry.Major);
+
+                if (result != 0)
+                {
+                    return result;
+                }
+
+                result = -Minor.CompareTo(otherEntry.Minor);
+
+                if (result != 0)
+                {
+                    return result;
+                }
+
+                result = -Patch.CompareTo(otherEntry.Patch);
+
+                if (result != 0)
+                {
+                    return result;
+                }
+            }
+
+            return base.InternalCompareTo(other);
+        }
+
+        private void SetVersion(string insertionText)
+        {
+            if (!string.IsNullOrEmpty(insertionText))
+            {
+                int atIndex = insertionText.IndexOf('@');
+
+                Debug.Assert(atIndex >= 0);
+
+                string versionText = insertionText.Substring(atIndex + 1);
+                string[] versionParts = versionText.Split('.');
```

Comment:
>string[] versionParts = versionText.Split('.'); [](start = 16, length = 47)

Would the System.Version class work for this instead of parsing out the parts yourself?

Reply:
Looks like I can use the SemanticVersion class.

---
In reply to: [198550380](https://github.com/aspnet/LibraryManager/pull/258#discussion_r198550380) [](ancestors = 198550380)